internal/worker/authenticationworker/utils.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package authenticationworker
+
+import (
+	"strings"
+
+	"github.com/juju/utils/v3/ssh"
+)
+
+const jujuEphemeralCommentPrefix = ssh.JujuCommentPrefix + "Ephemeral:"
+
+// ensureJujuEphemeralComment ensures that the key has a Juju ephemeral comment.
+// This is used to flush the ephemeral keys when the worker is restarted.
+func ensureJujuEphemeralComment(key string) string {
+	ak, err := ssh.ParseAuthorisedKey(key)
+	// Just return an invalid key as is.
+	if err != nil {
+		logger.Warningf("invalid Juju ssh key %s: %v", key, err)
+		return key
+	}
+	if ak.Comment == "" {
+		return key + " " + jujuEphemeralCommentPrefix + "sshkey"
+	} else {
+		// Add the Juju prefix to the comment if necessary.
+		if !strings.HasPrefix(ak.Comment, jujuEphemeralCommentPrefix) {
+			commentIndex := strings.LastIndex(key, ak.Comment)
+			return key[:commentIndex] + jujuEphemeralCommentPrefix + ak.Comment
```

Comment:
Can we please assign `key[:commentIndex]` to a variable, so that we can give it some meaning. This allows the reader to answer the question: What is `key[:commentIndex]`?